### PR TITLE
Standardize snake case convention for API keys

### LINKED ISSUE #381: Time dependent unit tests fail at 7:48 pm ET
FAIL  src/Utilities/useQueryParams.test.js
  Utilities/useQueryParams
    ✓ returns expected initial values as queryParams  (75 ms)
    ✓ returns setId, setStartDate, setEndDate, and setLimit as methods (7 ms)
    ✓ invoked methods returns new queryParams object (4 ms)
    ✓ setId, setLimit correctly handles null, undefined and NaN values (5 ms)
    ✕ setEndDate returns current day in `YYYY-MM-DD` string format (22 ms)
    ✕ setStartDate returns current day if passed a null value (19 ms)
    ✕ setStartDate returns expected day when invoked (29 ms)

### PR DESCRIPTION
**Pull request by benthomasson**
We have a convention of using snake case for API keys when making queries and when accessing data from queries.   This makes it clear that we are dealing with the API.  This PR removes keysToCamel and urlMappedQueryParams which change that convention to camel case. 

This PR also cleans up the eslint rules to make this convention the standard for all code in this project.

Also fixes #381

### CODE REVIEW

**Review by kialam — APPROVED** [review]

**Review by brumik — CHANGES_REQUESTED** [review]
This breaks the toolbar. If you try it out you will see that toolbar is unable to display values and send changes and nothing loads what is hooked up to the toolbar. The toolbar operated on the camel cased variables and options. If we need to remake the query params to camel case we need to remake most of the toolbar too.

**Comment by brumik** [comment]
Screenshot from 2020-12-17 16-45-51

**Comment by kialam** [comment]
@brumik check your db - it may not have past 30 days worth of data to show.

**Review by kialam — CHANGES_REQUESTED** [review]
Toolbar dropdown UI needs minor fix:
Screen Shot 2020-12-17 at 11 30 52 AM

**Comment by brumik** [comment]
I have data, and it works on devel

**Review by jctanner — APPROVED** [review]

File: src/Utilities/useQueryParams.test.js [hunk 1]
```diff
@@ -58,15 +59,15 @@ describe('Utilities/useQueryParams', () => {
     });
 
     it('setEndDate returns current day in `YYYY-MM-DD` string format', () => {
-        const currentDate = new Date().toJSON().slice(0, 10).replace(/-/g, '-');
+        const currentDate = moment(new Date().toISOString()).format('YYYY-MM-DD');
```
Comment (jctanner) [inline src/Utilities/useQueryParams.test.js:62]: :+1:

File: src/Utilities/useQueryParams.test.js [hunk 2]
```diff
@@ -58,15 +59,15 @@ describe('Utilities/useQueryParams', () => {
...
         act(() => {
             page.setEndDate();
         });
         expect(page.queryParams).toEqual({ ...initialValues, endDate: currentDate });
     });
 
     it('setStartDate returns current day if passed a null value', () => {
-        const currentDate = new Date().toJSON().slice(0, 10).replace(/-/g, '-');
+        const currentDate = moment(new Date().toISOString()).format('YYYY-MM-DD');
```
Comment (jctanner) [inline src/Utilities/useQueryParams.test.js:70]: :+1:

File: src/Utilities/useQueryParams.test.js [hunk 3]
```diff
@@ -76,7 +77,7 @@ describe('Utilities/useQueryParams', () => {
     it('setStartDate returns expected day when invoked', () => {
         const days = 8;
         const date = new Date(new Date().setDate(new Date().getDate() - days));
-        const expected = date.toJSON().slice(0, 10).replace(/-/g, '-');
+        const expected = moment(date.toISOString()).format('YYYY-MM-DD');
```
Comment (jctanner) [inline src/Utilities/useQueryParams.test.js:80]: :+1:

File: src/Containers/JobExplorer/JobExplorer.js [hunk 4]
```diff
@@ -53,67 +52,94 @@ const JobExplorer = ({
...
     useEffect(() => {
-        insights.chrome.appNavClick({ id: 'job-explorer', secondaryNav: true });
-
-        window.insights.chrome.auth.getUser()
-        .then(() =>
-            preflightRequest()
-            .catch((error) => { setPreFlightError({ preflightError: error }); })
-            // Loading is set false when the data also loaded
-        );
+        if (!isFirstRender) {
```
Comment (brumik) [inline src/Containers/JobExplorer/JobExplorer.js:80]: Can we revert this not to use the `isFirstRender` logic?

File: src/Containers/JobExplorer/JobExplorer.js [hunk 5]
```diff
@@ -53,67 +52,94 @@ const JobExplorer = ({
...
     useEffect(() => {
-        setApiError(null);
-        setIsLoading(true);
-        window.insights.chrome.auth.getUser()
-        .then(() => {
+        if (isFirstRender) {
+            return;
+        }
+
+        const update = async () => {
```
Comment (brumik) [inline src/Containers/JobExplorer/JobExplorer.js:123]: Missing guard `didCancel`

File: src/Containers/JobExplorer/JobExplorer.js [hunk 6]
```diff
@@ -53,67 +52,94 @@ const JobExplorer = ({
...
-        .then(() =>
-            preflightRequest()
-            .catch((error) => { setPreFlightError({ preflightError: error }); })
-            // Loading is set false when the data also loaded
-        );
+        if (!isFirstRender) {
+            return;
+        }
+
+        let ignore = false;
```
Comment (brumik) [inline src/Containers/JobExplorer/JobExplorer.js:84]: Can we call this variable `didCancel`? For me it explains more than `ignore` (it kinda answers the question: why we ignore saving)

**Review by brumik — CHANGES_REQUESTED** [review]
I think these JobExplorer API fetching changes are a bit off the scope of this PR. I made some request around the API loading, which we discussed earlier this week.

**Review by kialam — APPROVED** [review]

**Review by brumik — APPROVED** [review]

**Pull request merged**